lesson17/src/js/main.js

```diff
@@ -0,0 +1,225 @@
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const fragment = document.createDocumentFragment();
+  const renderLoadingBox = document.createElement("div");
+  renderLoadingBox.classList.add("loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const createTextBox = document.createElement("p");
+  createTextBox.classList.add("");
+  createTextBox.textContent = error;
+  slideImageArea.appendChild(createTextBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  const sliderArea = document.querySelector(".mainvisual__images");
+
+  sliderArea.appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) {
+      sliderItem.classList.add("is-active");
+    }
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
+
+  const buttonArea = document.createElement("div");
+  buttonArea.classList.add("slider__button");
+
+  const prevButton = document.createElement("button");
+  prevButton.setAttribute("class", "arrow prev");
+  prevButton.id = "js-button_prev";
+  prevButton.setAttribute("aria-label", "previous");
+  prevButton.textContent = "previous";
+
+  fragment.appendChild(prevButton);
+
+  const nextButton = document.createElement("button");
+  nextButton.setAttribute("class", "arrow next");
+  nextButton.id = "js-button_next";
+  nextButton.textContent = "next";
+  nextButton.setAttribute("aria-label", "next");
+
+  fragment.appendChild(nextButton);
+
+  const getSlideArea = document.querySelector(".mainvisual__images");
```

Comment:
IMO
I thought there should be no ‘get' in the variable name.

Reply:
Thank you for your review!
Certainly I thought that get was unnecessary for the variable, so I deleted it.
I fixed it.

[1cf8f9c](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/1cf8f9c032bca88e0de001c24945242b52847c41)